src/test/integration/controllers/finishedQuiz.spec.js

```diff
@@ -1,131 +1,153 @@
+
+const Quiz = require('~/models/quiz')
+
+const testUserAuthentication = require('~/utils/testUserAuth')
+const { UNAUTHORIZED, DOCUMENT_NOT_FOUND } = require('~/consts/errors')
+const {
+  roles: { TUTOR }
+} = require('~/consts/auth')
+const TokenService = require('~/services/token')
+
+const endpointUrl = '/finished-quizzes/'
+const nonExistingQuiz = '64cf8a3d40135fba5a0c8fa2'
+
+const testFinishedQuizData = {
+  grade: 100,
+  results: [
+    {
+      question: 'Is it the best programming language?',
+      answers: [
+        {
+          text: 'Yes',
+          isCorrect: true,
+          isChosen: false
+        },
+        {
+          text: 'Yes, of course',
+          isCorrect: false,
+          isChosen: true
+        }
+      ]
+    }
+  ]
+}
+
+const testQuizData = {
+  title: 'Assembly',
+  description: 'Description',
+  category: '6502ec2060ec37be943353e2',
+  items: ['6527ed6c14c6b72f36962364']
+}
+
+describe('Quiz controller', () => {
+  let app, server, accessToken, currentUser, testFinishedQuiz, testQuiz
+
+  beforeAll(async () => {
+    ;({ app, server } = await serverInit())
+  })
+
+  beforeEach(async () => {
+    accessToken = await testUserAuthentication(app, { role: TUTOR })
+
+    currentUser = TokenService.validateAccessToken(accessToken)
+
+    testQuiz = await Quiz.create({
+      author: currentUser.id,
+      ...testQuizData
+    })
+
+    testFinishedQuiz = await app
+      .post(endpointUrl)
+      .send({ quiz: testQuiz._id, ...testFinishedQuizData })
+      .set('Cookie', [`accessToken=${accessToken}`])
+  })
+
+  afterEach(async () => {
+    await serverCleanup()
+  })
+
+  afterAll(async () => {
+    await stopServer(server)
+  })
+
+  describe(`POST ${endpointUrl}`, () => {
+    it('should create a new finished quiz', async () => {
+      expect(testFinishedQuiz.statusCode).toBe(201)
+      expect(testFinishedQuiz._body).toMatchObject({
+        _id: expect.any(String),
+        createdAt: expect.any(String),
+        updatedAt: expect.any(String),
+        quiz: testQuiz._id,
+        ...testFinishedQuizData
+      })
+    })
+
+    it('should throw UNAUTHORIZED', async () => {
+      const response = await app.post(endpointUrl)
+
+      expectError(401, UNAUTHORIZED, response)
+    })
+
+    it('should throw DOCUMENT_NOT_FOUND for quiz', async () => {
+      const response = await app
+        .post(endpointUrl)
+        .send({
+          ...testFinishedQuizData,
+          quiz: nonExistingQuiz
+        })
+        .set('Cookie', [`accessToken=${accessToken}`])
+
+      expectError(404, DOCUMENT_NOT_FOUND([Quiz.modelName]), response)
+    })
+  }),
+    describe(`GET ${endpointUrl}`, () => {
+      it('should get all finished quizzes', async () => {
+        const response = await app.get(`${endpointUrl}`).set('Cookie', [`accessToken=${accessToken}`])
+
+        expect(response.statusCode).toBe(200)
+        expect(Array.isArray(response.body.items)).toBeTruthy()

```

Comment:
You don't need to check `true/false` (isArray returns boolean) for truthy value
```suggestion
        expect(Array.isArray(response.body.items)).toBe(true)

```

Reply:
Same here, it's not the endpoint that I wrote, but I can change it, no problem.

Reply:
Same here, it's not the endpoint test that I wrote, but anyway I can fix it, no problem.

Reply:
Check [this](https://stackoverflow.com/questions/1967370/git-replacing-lf-with-crlf), plobably you need to run one of git commands (`git config core.autocrlf...`) to fix that. End of line should not be triggered as changes.

Reply:
Fixed.

Reply:
Great, thank you!